frontend/src/pages/projects/notebook/utils.ts

```diff
@@ -252,3 +253,17 @@ export const useNotebookStatus = (
 
 export const getEventFullMessage = (event: EventKind): string =>
   `${getEventTimestamp(event)} [${event.type}] ${event.message}`;
+
+export const checkNotebooksRestart = (
```

Comment:
```suggestion
export const willNotebooksRestart = (
```

Any reason you didn't make this a hook? It seems unnecessary to request the caller to pass you the `notebookStates` when it's a context variable.

Reply:
Yeah I made it a hook at first, but when using it, I found some scenarios in which I cannot use a hook... but now when I go back and check I find that the problem is solved. So I will change it to a hook.